File: backend-node-typescript/src/features/continent/continent.service.ts
import { injectable } from 'tsyringe';
import { ContinentRepository } from './continent.repository';
import { Continent } from './continent.model';

@injectable()
export class ContinentService {
  constructor(private continentRepository: ContinentRepository) {}

  getAllContinents(): Promise<Continent[]> {
    return this.continentRepository.findAll();
  }

}

File: backend-springboot/src/main/java/com/ganatan/backend_java/modules/continent/ContinentService.java
package com.ganatan.backend_java.modules.continent;

import org.springframework.stereotype.Service;

import java.util.List;
import org.springframework.aop.framework.AopProxyUtils;

@Service
public class ContinentService {

	private final ContinentRepositoryInterface repository;

	public ContinentService(ContinentRepositoryInterface repository) {
		System.out.println("[ganatan] ContinentService Constructor:");
		System.out.println(
				"[ganatan] Repository injecté : " + AopProxyUtils.ultimateTargetClass(repository).getSimpleName());
		this.repository = repository;
	}

	public List<Continent> getItems() {
		System.out.println("[ganatan] ContinentService getItems:");
		return repository.findAll();
	}
}
